backend/endpoints/threads.py

```diff
@@ -168,42 +184,33 @@ async def update_threads_message(
     org_id: str,
     room_id: str,
     message_id: str,
+    thread_id: str,
     request: MessageRequest,
     background_tasks: BackgroundTasks,
 ):
+    
+    message = await get_message(org_id, room_id, message_id)
 
-    payload = request.dict()
-    loadedMessage = await get_message(org_id, room_id, message_id)
-    loadedMessageThread = loadedMessage['threads']
-    thread_to_add = {"threads": [payload, *loadedMessageThread]}
-
-    if not loadedMessage:
+    if not message:
         raise HTTPException(
             status_code=status.HTTP_404_NOT_FOUND, detail="Message not found"
         )
 
-    added_thread_message = await edit_message_threads(org_id, message_id, thread_to_add)
-
-    if not added_thread_message:
-        raise HTTPException(
-            status_code=status.HTTP_424_FAILED_DEPENDENCY,
-            detail={"thread not added": added_thread_message},
-        )
+    payload = request.dict(exclude_unset=True)
 
-    # payload["edited"] = True
-    loadedMessage.update(thread_to_add)
+    edited_thread_message = await edit_thread_message(org_id, message_id, thread_id, payload, message)
```

Comment:
This variable is unused...
You should check whether the thread msg has been successfully edited or not.